Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2233,8 +2227,18 @@ void WebProcessProxy::didStartProvisionalLoadForMainFrame(const URL& url)
         return;
     }
 
-    // Associate the process with this site.
-    m_site = WTFMove(site);
+    if (siteIsolationEnabled)
```

Comment:
You can just use `m_sharedPreferencesForWebProcess.siteIsolationEnabled` and remove the parameter.